server/src/routes/bookings.ts

```diff
@@ -34,3 +37,123 @@ bookingRoutes.post(
+    return;
+  }
+
+  // get all dates already booked for that listing
+  const futureBookingsForListing = [];
+  const bookings = await bookingsData.getBookingsByListingId(listingId);
+  bookings.forEach((booking) => {
+    const startDateJustDate = new Date(booking.startTimestamp.toDate().getFullYear(), booking.startTimestamp.toDate().getMonth(), booking.startTimestamp.toDate().getDate());
+    const endDateJustDate = new Date(booking.endTimestamp.toDate().getFullYear(), booking.endTimestamp.toDate().getMonth(), booking.endTimestamp.toDate().getDate());
+
+    if (endDateJustDate > new Date(today.getFullYear(), today.getMonth(), today.getDate())) {
+      let isUnique = futureBookingsForListing.find(date => date.getTime() === startDateJustDate.getTime());
+
+      if(!isUnique) {
+        futureBookingsForListing.push(startDateJustDate);
+      } else if (futureBookingsForListing.length === 0) {
+        futureBookingsForListing.push(startDateJustDate);
+      }
+
+      let nextDateJustTime = new Date(startDateJustDate.getFullYear(), startDateJustDate.getMonth(), startDateJustDate.getDate());
+      while (endDateJustDate.getTime() > nextDateJustTime.getTime()) {
+        nextDateJustTime.setDate(nextDateJustTime.getDate() + 1);
+        let isUnique = futureBookingsForListing.find(date => date.getTime() === nextDateJustTime.getTime());
+        
+        if(!isUnique){
+          futureBookingsForListing.push(new Date(nextDateJustTime.getFullYear(), nextDateJustTime.getMonth(), nextDateJustTime.getDate())); 
+        }
+      }
+    }
+  });
+
+  // get all dates wanting to be booked
+  let datesTryingToBook = [startDateJustDate];
+  let nextDateJustDate = new Date(startDateJustDate.getFullYear(), startDateJustDate.getMonth(), startDateJustDate.getDate());
+  while (endDateJustDate.getTime() > nextDateJustDate.getTime()) {
+    nextDateJustDate.setDate(nextDateJustDate.getDate() + 1);
+    let isUnique = datesTryingToBook.find(date => date.getTime() === nextDateJustDate.getTime());
+
+    if (!isUnique) {
+      datesTryingToBook.push(new Date(nextDateJustDate.getFullYear(), nextDateJustDate.getMonth(), nextDateJustDate.getDate()));
+    }
+  }
+
+  // check if dates booked conflict with dates being tried to book
+  try {
+    datesTryingToBook.forEach((date) => {
+      futureBookingsForListing.forEach((futureBooking) => {
+        if (date.getTime() === futureBooking.getTime()) {
+          throw 'Dates to be book conflict';
+        }
+      });
+    });
+  } catch (e) {
+    res.status(409).json({ message : e});
+    return;
+  }
+  
+
+  const diffInDays = Math.round((endDateJustDate.getTime() - startDateJustDate.getTime()) / (1000 * 60 * 60 * 24));
+  const booking = await bookingsData.createBooking(bookerId, listingId, numOfPeople, listingFound.owner, (listingFound.price * (diffInDays + 1)), endDateJustDate.toString(), startDateJustDate.toString());
```

Comment:
listingFound.owner should be listingFound.ownerId

Reply:
listing has an owner field not an ownerId field